src/edgeDebugAdapter.ts

```diff
@@ -109,25 +109,33 @@ export class EdgeDebugAdapter extends CoreDebugAdapter {
                 edgeArgs.push(launchUrl);
             }
 
-            this._edgeProc = this.spawnEdge(runtimeExecutable, edgeArgs, edgeEnv, edgeWorkingDir, !!args.runtimeExecutable);
-            this._edgeProc.on('error', (err) => {
-                const errMsg = 'Chrome error: ' + err;
+            let launchResult: Promise<ISpawnHelperResult>;
+
+            [this._edgeProc, launchResult] = this.spawnEdge(runtimeExecutable, edgeArgs, edgeEnv, edgeWorkingDir, !!args.runtimeExecutable);
+            const erroHanlder = (errMessage: string) => {
+                const errMsg = 'Edge launch error: ' + errMessage;
                 logger.error(errMsg);
                 this.terminateSession(errMsg);
-            });
+            };
 
-            return args.noDebug ? undefined :
-                this.doAttach(port, launchUrl || args.urlFilter, args.address, args.timeout, undefined, args.extraCRDPChannelPort)
-                .then(() => {
-                    this._scriptParsedEventBookKeeping = {};
-                    this._debugProxyPort = port;
+            this._edgeProc.on('error', erroHanlder);
 
-                    if (!this._chromeConnection.isAttached || !this._chromeConnection.attachedTarget) {
-                        throw coreUtils.errP(localize("edge.debug.error.notattached", "Debugging connection is not attached after the attaching process."));
-                    }
+            return launchResult.catch((err: Error) => {
+                erroHanlder(err.message);
+                throw err;
+            }).then(() => {
```

Comment:
Do you think that refactoring this code to use async/await would make this code clearer?

Reply:
Sure. I can do that.